src/estimators/exact_set.py

```diff
@@ -53,12 +55,20 @@ def __contains__(self, x):
 
   def add(self, x):
     """Adds an id x to the sketch."""
-    self._ids.add(x)
+    self._ids[x] = self._ids.get(x, 0) + 1
 
   def ids(self):
     """Return the internal ID set."""
     return self._ids
 
+  def frequency(self, x):
+    """Returns the frequency of occurrence of x."""
+    return self._ids.get(x, 0)
+
+
+class ExactSet(ExactMultiSet):
```

Comment:
The semantics of this inheritance are weird.

I would not expect to be able to pass as ExactSet anywhere an ExactMultiSet is required. I also would not expect "ExactSet" to have a "frequency" method.

It saves some code like this, but deriving from SketchBase, aggregating an ExactMultiSet and forwarding the methods would be cleaner.

Reply:
I guess I subscribe to the philosophy that twice as much code equals twice as many opportunities for undetected bugs and twice as much maintenance headache.  But I agree with the point of view that if ExactSet has a frequency method, it should report something that is appropriate for a set.  Let me know if you are OK with these changes.

Reply:
Per offline conversation we agreed to simply replace ExactSet with ExactMultiSet.